test/e2e/data/e2e_conf.yaml

```diff
@@ -169,10 +169,10 @@ variables:
   EXP_CLUSTER_RESOURCE_SET: "true"
   EXP_MACHINE_POOL: "true"
   CLUSTER_TOPOLOGY: "true"
-  INIT_WITH_BINARY_V1BETA1: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.2.0/clusterctl-{OS}-{ARCH}"
+  INIT_WITH_BINARY_V1BETA1: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.4.0/clusterctl-{OS}-{ARCH}"
```

Comment:
Made changes to clusterctl upgrade CAPI test according to [this](https://github.com/kubernetes-sigs/cluster-api/pull/8518#issuecomment-1508064859) change, but still the tests are failing.
@killianmuldoon can you help here?
cc @richardcase